docs/generate_docs.py

```diff
@@ -3,40 +3,56 @@
     Generate this these files into the directory docs/source with the gen_ prefix.
     These will be included bv the hand referenced .vst files in the same directory.
 """
-# pylint: disable=E0401,C0413,C0103,C0301,W1514,R0914,W1309
+
+# pylint: disable=E0401,C0413,C0103,C0301,W1514,R0914,W1309,W0212,R0915,R0912,E1121
 import sys
 import os
 import json
 import yaml
-import generate_docs_options
 
 sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "../src")))
-from hf_hydrodata.data_model_access import load_data_model
+from hf_hydrodata.data_model_access import load_data_model, ModelTableRow
 
 
 def main():
     """
     Main function for the docs generator.
     """
 
-    load_data_model(load_from_api=False)
+    load_data_model()
+    load_rows()
     generate_datasets()
 
 
+def load_rows():
+    """Load all rows from data model for table used to generate read-the-docs documentation."""
+    data_model = load_data_model()
+    for table_name in [
+        "dataset",
+        "grid",
+        "temporal_resolution",
+        "data_catalog_entry",
+        "dataset_type",
+    ]:
+        table_object = data_model.get_table(table_name)
+        result_json = table_object._query_data_catalog({})
+        for row_id in result_json:
+            if not row_id in table_object.row_ids:
+                table_object.row_ids.append(row_id)
+                response_row = result_json.get(row_id)
+                if response_row is not None:
+                    row = ModelTableRow(response_row)
+                    table_object.rows[row_id] = row
+
+
 def generate_datasets():
     """Generate the documentation of datasets"""
 
     directory = os.path.abspath(os.path.join(os.path.dirname(__file__), "source"))
-    (
-        dataset_type_ids,
-        dataset_ids,
-        variable_ids,
-        grid_ids,
-        temporal_resolution_ids,
-    ) = _collect_visible_ids()
+    (dataset_type_ids, dataset_ids, variable_ids) = _collect_visible_ids()
 
     gen_dataset_list_path = f"{directory}/gen_dataset_list.rst"
-    with open(gen_dataset_list_path, "w+") as stream:
+    with open(gen_dataset_list_path, "w+"):
         pass
```

Comment:
This is creating the file as an empty file. Other functions open the file and append to it later.
I added a comment to say this. This may not be necessary since in a real model there would always be some datasets. I left it in since it didn't hurt. It was like initializing a variable before using it.